instances/etcd/main.tf

```diff
@@ -10,7 +10,13 @@ resource "oci_core_instance" "TFInstanceEtcd" {
   hostname_label      = "${var.hostname_label}-${count.index}"
   image               = "${lookup(data.oci_core_images.ImageOCID.images[0], "id")}"
   shape               = "${var.shape}"
-  subnet_id           = "${var.subnet_id}"
+
+  create_vnic_details {
+    subnet_id         = "${var.subnet_id}"
+    display_name      = "${var.label_prefix}${var.display_name}-${count.index}"
+    hostname_label    = "${var.hostname_label}-${count.index}"
+    private_ip        = "${var.assign_private_ip == "true" ? cidrhost(lookup(var.network_cidrs,var.subnet_name), count.index+5) : ""}"
```

Comment:
@srirg what's your rationale for picking the `count.index+5` IP in the etcd subnet?

Reply:
As you know the first 2 IPs and the last one in a subnet are reserved for OCI. Also when the lb-etcd private is enabled, its picking up count.index+2, so using count.index+5. Also we could make this variable so it can changed easily as well.

Reply:
OK, if that's the case, shouldn't the instance IP be count.index + _3_ not 5?

i.e. if`etcdSubnetAD1` / `10.0.20.0/24` looks like:

```
[0] 10.0.20.0 => subnet
[1] 10.0.20.1 => reserved for OCI
[2] 10.0.20.2 => reserved for OCI
[3] 10.0.20.3 => reserved fo etcd LB (if enabled)
[4] 10.0.20.4 => first free IP
[5] 10.0.20.5 => second free IP
[6] 10.0.20.6 => third free IP
...
```

then, oci_core_instance.TFInstanceEtcd[1] in etcdSubnetAD1 should be assigned `10.0.20.4` from `cidrhost("10.0.20.0/24", 1 + 3)`.

Also remember that `etcdSubnetAD2` is has its own block `10.0.21.0/24` and its own `count`. 

Let me know if I am making a bad assumption.

Reply:
Just a small correction,

[0] 10.0.20.0 => reserved for OCI
[1] 10.0.20.1 => reserved for OCI
[2] 10.0.20.2 =>  reserved fo etcd LB (if enabled)
[3] 10.0.20.3 =>first free IP

 that's correct we could start with .3 as well but when I tested without assigning an ip and etcd private lbr enabled it started from .5  so just to be symmetric I put it 5 as the index is 0 for the 1st count.